optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/PillarChangeMove.java

```diff
@@ -125,28 +123,22 @@ public String getSimpleMoveTypeDescription() {
     }
 
     @Override
-    public boolean equals(Object o) {
+    public boolean equals(final Object o) {
         if (this == o) {
             return true;
-        } else if (o instanceof PillarChangeMove) {
-            PillarChangeMove<?> other = (PillarChangeMove) o;
-            return new EqualsBuilder()
-                    .append(variableDescriptor, other.variableDescriptor)
-                    .append(pillar, other.pillar)
-                    .append(toPlanningValue, other.toPlanningValue)
-                    .isEquals();
-        } else {
+        }
+        if (o == null || getClass() != o.getClass()) {
```

Comment:
+1 to doing the getClass check instead of instanceof. Especially with ChainedChangeMove and ChangeMove it might matter some day.